### PR TITLE
jdk9

### PR DESCRIPTION
**Pull request by stevenschlansker**
This seems to get Travis to test on jdk9.
Right now it fails due to a Kotlin access error.

### CODE REVIEW

**Comment by arteam** [comment]
The Kotlin plugin seems to fail miserably on JDK9 because of machinations with `sun.misc.Unsafe`.

**Comment by arteam** [comment]
We can try to upgrade the version of the Kotlin runtime in the plugin. If I remember correctly, there should be support for JDK9/10 in the latest versions.

**Comment by arteam** [comment]
https://blog.jetbrains.com/kotlin/2017/08/kotlin-1-1-4-is-out/

**Comment by stevenschlansker** [comment]
I think @qualidafial said he's working on a jdk9 branch so I don't want to make too many changes until I see what he's already done.

**Comment by qualidafial** [comment]
My PR is merged, and I just merged latest from master into this branch--we'll see what difference it makes to the build.

File: .travis.yml [hunk 1]
```diff
@@ -31,6 +31,14 @@ matrix:
         #- '[ "$TRAVIS_SECURE_ENV_VARS"  = "true" ] || mvn -Dmaven.javadoc.skip=true -Dbasepom.check.skip-basic=true -Dbasepom.check.skip-findbugs=true -Dbasepom.check.skip-pmd=true -Dbasepom.check.skip-checkstyle=true -Ptoolchains        -B verify'
         - mvn -Dmaven.javadoc.skip=true -Dbasepom.check.skip-basic=true -Dbasepom.check.skip-findbugs=true -Dbasepom.check.skip-pmd=true -Dbasepom.check.skip-checkstyle=true -Ptoolchains -B verify
       jdk: oraclejdk8
+    - env:
+        - TESTS=Y
+      script:
+        # Oracle profile disabled until they fix their orai18n JAR artifact on maven.oracle.com repository.
+        #- '[ "$TRAVIS_SECURE_ENV_VARS" != "true" ] || mvn -Dmaven.javadoc.skip=true -Dbasepom.check.skip-basic=true -Dbasepom.check.skip-findbugs=true -Dbasepom.check.skip-pmd=true -Dbasepom.check.skip-checkstyle=true -Djdk9=true -Ptoolchains,oracle -B verify'
```
Comment (stevenschlansker) [inline .travis.yml]: can we factor all this junk out into matrix better?  it sucks to have to copy+paste it around, but I'm not super comfortable with the build matrix to know how to do better.

Comment (qualidafial) [reply .travis.yml]: I don't know why we're not just putting these commands in bash scripts and having our matrix just invoke those. This "fit everything on one line" thing makes it really hard to read anything.

Comment (stevenschlansker) [reply .travis.yml]: That makes sense to me.  I have a pretty busy week but I'll take a crack at it sometime unless someone else gets to it first.

**Comment by stevenschlansker** [comment]
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project jdbi3-core: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test failed: java.lang.NoClassDefFoundError: java/sql/SQLException: java.sql.SQLException -> [Help 1]
```
wut

**Comment by qualidafial** [comment]
Gotta go. We may need to explicitly add a dep on java.sql module in jdk9 builds to resolve that NoClassDefFoundError

**Comment by stevenschlansker** [comment]
surefire 2.20.1 includes the modules for us, but we still hit more errors later.

**Comment by stevenschlansker** [comment]
Someone else is trying it too, and having troubles, sounds like it may be a work in progress upstream https://stackoverflow.com/questions/49580261/kotlin-java-10-maven-java-modules-fail

**Comment by FredDeschenes** [comment]
FYI I checked out this branch and all the tests ran successfully on Linux with OpenJDK 9.0.4

**Comment by qualidafial** [comment]
FYI, the latest build is failing not on Kotlin compile or test, but on the license header check.

**Comment by qualidafial** [comment]
BAZINGA!!

**Pull request merged**